cloudinit/tests/test_features.py

```diff
@@ -0,0 +1,57 @@
+"""
+This file is for testing the feature flag functionality itself,
+NOT for testing that any individual feature flag
+"""
+import pytest
+from importlib import reload
+from pathlib import Path
+
+import cloudinit
+import cloudinit.features
+
+
+@pytest.yield_fixture()
+def create_override(request):
+    override_path = Path(cloudinit.__file__).parent / 'feature_overrides.py'
+    with override_path.open('w') as f:
+        for key, value in request.param.items():
+            f.write('{} = {}\n'.format(key, value))
+
+    # The goal is to reload features.py so we also get the
+    # features_overrides.py stuff. This could normally be achieved with
+    # a single reload(cloudinit.features), however, since this fixture will be
+    # called multiple times in the same interpreter, calling
+    # reload(cloudinit.features) a second time won't refresh what already exists
+    # in feature_overrides.py because that's now already defined, so we need
+    # to call reload(cloudinit.feature_overrides) first. But the first time
+    # we call this fixture, cloudinit.feature_overrides doesn't exist yet, so
+    # we need to reload cloudinit.features first before we can reload
+    # cloudinit.feature_overrides, hence the three calls here.
+    reload(cloudinit.features)
```

Comment:
wow thanks for this. I've never played with importlib directly. Great contextual comments